src/main/kotlin/no/nav/syfo/sykmelding/service/EgenmeldtSykmeldingService.kt

```diff
@@ -80,65 +75,48 @@ class EgenmeldtSykmeldingService @KtorExperimentalAPI constructor(
+        EGENMELDT_SYKMELDING_COUNTER.inc()
     }
 
-    private fun opprettReceivedSykmelding(pasient: Pasient, sykmeldingId: String, fellesformat: XMLEIFellesformat): ReceivedSykmelding {
-        val fellesformatMarshaller: Marshaller = JAXBContext.newInstance(XMLEIFellesformat::class.java, XMLMsgHead::class.java, HelseOpplysningerArbeidsuforhet::class.java).createMarshaller()
-            .apply { setProperty(Marshaller.JAXB_ENCODING, "UTF-8") }
-
-        val healthInformation = extractHelseOpplysningerArbeidsuforhet(fellesformat)
-        val msgHead = fellesformat.get<XMLMsgHead>()
-
-        val sykmelding = healthInformation.toSykmelding(
-            sykmeldingId = sykmeldingId,
-            pasientAktoerId = pasient.aktorId,
-            legeAktoerId = pasient.aktorId,
-            msgId = sykmeldingId,
-            signaturDato = msgHead.msgInfo.genDate
-        )
-        return ReceivedSykmelding(
-            sykmelding = sykmelding,
-            personNrPasient = pasient.fnr,
-            tlfPasient = healthInformation.pasient.kontaktInfo.firstOrNull()?.teleAddress?.v,
-            personNrLege = pasient.fnr,
-            navLogId = sykmeldingId,
-            msgId = sykmeldingId,
-            legekontorOrgNr = "889640782",
-            legekontorOrgName = "NAV",
-            legekontorHerId = null,
-            legekontorReshId = null,
-            mottattDato = LocalDateTime.now(),
-            rulesetVersion = healthInformation.regelSettVersjon,
-            fellesformat = fellesformatMarshaller.toString(fellesformat),
-            tssid = dummyTssIdent
-        )
+    suspend fun validerEgenmeldtSykmelding(fom: LocalDate, tom: LocalDate, harFortroligAdresse: Boolean, fnr: String, userToken: String, callId: String) {
+        if (tom.isBefore(fom)) {
+            log.warn("Tom-dato er før fom-dato for callId {}", callId)
+            EGENMELDT_SYKMELDING_ERROR_TOM_BEFORE_FOM_COUNTER.inc()
+            throw TomBeforeFomDateException("Tom-dato er før fom-dato")
+        }
+        if (harFortroligAdresse) {
+            log.warn("Bruker har ikke tilgang til tjenesten, callId {}", callId)
+            throw IkkeTilgangException("Bruker har ikke tilgang til tjenesten")
+        }
+        if (fom.isBefore(tidligsteGyldigeFom)) {
+            log.warn("Egenmeldt sykmelding er ikke tilgjengelig før {}", tidligsteGyldigeFom)
+            throw ForTidligsteFomException("Egenmeldt sykmelding er ikke tilgjengelig før")
+        }
+        if (tom.isAfter(fom.plusDays(maxAntallDagerSykmeldt.toLong()))) {
+            log.warn("Egenmeldt sykmelding kan ikke være mer enn {} dager", maxAntallDagerSykmeldt)
+            throw ForLangPeriodeException("Egenmeldt sykmelding kan ikke være lenger enn $maxAntallDagerSykmeldt dager")
+        }
+        if (database.sykmeldingErAlleredeRegistrert(fnr = fnr, periode = Periode(fom = fom, tom = tom))) {
```

Comment:
Her sjekker jeg kun om det finnes en sykmelding på samme bruker for samme periode fra før. Jeg har ikke helt landet på om jeg må ha med arbeidsforhold eller ikke her, skal sove på saken.. :)